homeassistant/components/srp_energy/sensor.py

```diff
@@ -68,78 +88,67 @@ async def async_update_data():
 
     coordinator = DataUpdateCoordinator(
         hass,
-        _LOGGER,
+        LOGGER,
         name="sensor",
         update_method=async_update_data,
         update_interval=MIN_TIME_BETWEEN_UPDATES,
     )
 
     # Fetch initial data so we have data when entities subscribe
-    await coordinator.async_refresh()
+    await coordinator.async_config_entry_first_refresh()
 
-    async_add_entities([SrpEntity(coordinator)])
+    async_add_entities([SrpEntity(coordinator, entry_name)])
 
 
 class SrpEntity(SensorEntity):
     """Implementation of a Srp Energy Usage sensor."""
 
+    _attr_has_entity_name = True
```

Comment:
```suggestion
```

Reply:
I need to keep the name to be consistent with the previous implementation.

Reply:
For now, just keep the old code around `name` and don't add `_attr_has_entity_name` now. It's better to do that separately as more things need to change to accomplish that.